mern-ecommerce/controllers/order.js

```diff
@@ -61,3 +61,15 @@ exports.updateOrderStatus = (req, res) => {
     }
   );
 };
+
+exports.purgeOrder = async (req, res) => {
+  try {
+    const result = await Order.findOneAndRemove({ _id: req.params.orderId });
+    if (result) {
```

Comment:
In the `exports.purgeOrder` function, the condition `if (result) {` is incorrect. It should be `if (!result) {` because `findOneAndRemove` returns `null` if no document is found, and the message "Order not located" should be returned in this case.